ee/tabby-db/src/email_service_credential.rs

```diff
@@ -40,8 +40,22 @@ impl DbConn {
 
     pub async fn update_email_service_credential(
         &self,
-        creds: EmailServiceCredentialDAO,
+        smtp_username: String,
+        smtp_password: Option<String>,
+        smtp_server: String,
     ) -> Result<()> {
+        let smtp_password = match smtp_password {
+            Some(pass) => pass,
+            None => {
+                self.read_email_service_credential()
```

Comment:
would it make sense to execute this query inside of call block to make it a transaction? so there's a gurantee of atomic for write-after-read